Data/Vector/Generic.hs

```diff
@@ -1455,6 +1455,12 @@ findIndex :: Vector v a => (a -> Bool) -> v a -> Maybe Int
 {-# INLINE findIndex #-}
 findIndex f = Bundle.findIndex f . stream
 
+-- | /O(n)/ Yield 'Just' the index of the /last/ element matching the predicate
+-- or 'Nothing' if no such element exists.
+findIndexR :: Vector v a => (a -> Bool) -> v a -> Maybe Int
+{-# INLINE findIndexR #-}
```

Comment:
I'm actually not sure if it makes sense to inline this. [dolio remarked](https://github.com/haskell/vector/issues/172) that this function won't be able to use any stream fusion. I know too little about vector's optimisations to judge this.